File: Find First Palindromic String in the Array.py

# Solution 1
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:

        for word in words:
            if word == word[::-1]:
                return word

        return ""


# Solution 2
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        
        for word in words:
            left = 0
            right = len(word) - 1

            while word[left] == word[right]:
                left += 1
                right -= 1
                if left >= right:
                    return word

        return ""

File: Find First Palindromic String in the Array.java

// solution 1
class Solution {
    public String firstPalindrome(String[] words) {
        for (String word : words) {
            if (word.equals(new StringBuilder(word).reverse().toString())) {
                return word;
            }
        }
        return "";        
    }
}

// solution 2
class Solution {
    public String firstPalindrome(String[] words) {
        for (String word : words) {
            int left = 0;
            int right = word.length() - 1;
            
            while (left < right) {
                if (word.charAt(left) != word.charAt(right)) {
                    break;
                }
                left++;
                right--;
            }
            if (left >= right) {
                return word;
            }
        }
        return "";        
    }
}
